example.jsonnet

```diff
@@ -1,12 +1,16 @@
 local k = import "ksonnet.beta.3/k.libsonnet";
+local service = k.core.v1.service;
+local servicePort = k.core.v1.service.mixin.spec.portsType;
 local grafana = import "grafana/grafana.libsonnet";
-local namespace = "monitoring";
+local namespace = "monitoring-grafana";
 
 k.core.v1.list.new([
     grafana.dashboardDefinitions.new(namespace),
     grafana.dashboardSources.new(namespace),
     grafana.dashboardDatasources.new(namespace),
     grafana.deployment.new(namespace),
     grafana.serviceAccount.new(namespace),
-    grafana.service.new(namespace),
+    grafana.service.new(namespace) +
+        service.mixin.spec.withPorts(servicePort.newNamed("http", 3000, "http") + servicePort.withNodePort(30902)) +
```

Comment:
Can't we move those mixins into the service file? Or is this an example of overriding the service?

Reply:
1) It's an example of overriding
2) I think the sensible default for a package is not to expose a NodePort, people are surely going to get bitten by it if we make that the default of the package.